app/services/tag_following_service.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class TagFollowingService
+  def initialize(user=nil)
+    @user = user
+  end
+
+  def create(name)
+    name_normalized = ActsAsTaggableOn::Tag.normalize(name)
+    raise ArgumentError, "Name field null or empty" if name_normalized.blank?
+
+    tag = ActsAsTaggableOn::Tag.find_or_create_by(name: name_normalized)
+    tag_following = @user.tag_followings.new(tag_id: tag.id)
+
+    raise "Can't process tag entity" unless tag_following.save
+    tag
+  end
+
+  def destroy(id)
+    tag_following = @user.tag_followings.find_by(tag_id: id)
+    tag_following&.destroy
```

Comment:
It's the safe navigation operator so should be replacing logic like "tag_following && tag_following.destroy"  While all the unit tests pass thing may need to look into is if it should be returning false instead of nil or something like that, but with all the unit tests passing even for error conditions I thought it was done correctly.

Reply:
That's great. Thank you for the explaination. I didn't know that operator.